### PR TITLE
crd openapi e2e: increase waitForOpenAPISchema timeout 

### LINKED ISSUE #86418: [Flaky test] [sig-api-machinery] CustomResourcePublishOpenAPI flakes
**Which jobs are flaking**:
pull-kubernetes-e2e-gce

**Which test(s) are flaking**:

- [sig-api-machinery] CustomResourcePublishOpenAPI [Privileged:ClusterAdmin] updates the published spec when one version gets renamed [Conformance]
- [sig-api-machinery] CustomResourcePublishOpenAPI [Privileged:ClusterAdmin] works for CRD without validation schema [Conformance]
- [sig-api-machinery] CustomResourcePublishOpenAPI [Privileged:ClusterAdmin] works for multiple CRDs of same group but different versions [Conformance]
- [sig-api-machinery] CustomResourcePublishOpenAPI [Privileged:ClusterAdmin] works for multiple CRDs of same group and version but different kinds [Conformance]
- [sig-api-machinery] CustomResourcePublishOpenAPI [Privileged:ClusterAdmin] works for CRD preserving unknown fields at the schema root [Conformance]
- [sig-api-machinery] CustomResourcePublishOpenAPI [Privileged:ClusterAdmin] works for multiple CRDs of different groups [Conformance]
- [sig-api-machinery] CustomResourcePublishOpenAPI [Privileged:ClusterAdmin] works for CRD preserving unknown fields in an embedded object [Conformance]
- [sig-api-machinery] CustomResourcePublishOpenAPI [Privileged:ClusterAdmin] works for CRD with validation schema [Conformance]
- [sig-api-machinery] CustomResourcePublishOpenAPI [Privileged:ClusterAdmin] removes definition from spec when one version gets changed to not be served [Conformance]

**Testgrid link**:
https://testgrid.k8s.io/presubmits-kubernetes-blocking#pull-kubernetes-e2e-gce&include-filter-by-regex=api-machinery&width=5&sort-by-flakiness=200

**Reason for failure**:
All the failures seem to be timeouts waiting for openapi publishing changes to take effect

/assign @roycaihw @sttts

### PR DESCRIPTION
**Pull request by roycaihw**
**What type of PR is this?**
/kind flake

**What this PR does / why we need it**:
Increase the timeout and add small improvements to deflake the test. Here is an issue that summarizes the problem and tracks potential improvements to the pipeline: https://github.com/kubernetes/kubernetes/issues/86967

**Which issue(s) this PR fixes**:

xref #86418

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/sig api-machinery
/cc @liggitt @sttts

### CODE REVIEW

File: test/e2e/apimachinery/crd_publish_openapi.go [hunk 1]
```diff
@@ -596,12 +596,12 @@ func waitForDefinitionCleanup(c k8sclientset.Interface, name string) error {
...
 func waitForOpenAPISchema(c k8sclientset.Interface, pred func(*spec.Swagger) (bool, string)) error {
-	client := c.CoreV1().RESTClient().(*rest.RESTClient).Client
-	url := c.CoreV1().RESTClient().Get().AbsPath("openapi", "v2").URL()
+	client := c.Discovery().RESTClient().(*rest.RESTClient).Client
+	url := c.Discovery().RESTClient().Get().AbsPath("openapi", "v2").URL()
 	lastMsg := ""
 	etag := ""
 	var etagSpec *spec.Swagger
-	if err := wait.Poll(500*time.Millisecond, wait.ForeverTestTimeout, mustSucceedMultipleTimes(waitSuccessThreshold, func() (bool, error) {
+	if err := wait.Poll(500*time.Millisecond, 45*time.Second, mustSucceedMultipleTimes(waitSuccessThreshold, func() (bool, error) {
```
Comment (liggitt) [inline test/e2e/apimachinery/crd_publish_openapi.go]: if we were exceeding 30 seconds regularly, I would double this to 1 minute and track improving the performance in an issue

Comment (roycaihw) [reply test/e2e/apimachinery/crd_publish_openapi.go]: changed to 1 minute and opened the issue: #86967

**Comment by liggitt** [comment]
/priority important-soon
this will alleviate CI flakes, will keep the linked test failure issue open to track improving this component in this milestone

**Pull request merged**